oneview/resource_ethernet_network.go

```diff
@@ -202,6 +267,32 @@ func resourceEthernetNetworkUpdate(d *schema.ResourceData, meta interface{}) err
 		return fmt.Errorf("Initial scope uri can not be updated")
 	}
 
+	if d.HasChange("bandwidth") {
+		rawVal := d.Get("bandwidth")
+		bandwidthVal := rawVal.([]interface{})
+		for _, bandwidth := range bandwidthVal {
+			rawBandwidth := bandwidth.(map[string]interface{})
+			conTempURI := utils.NewNstring(d.Get("connection_template_uri").(string))
+			// get connection template by uri
+			conTemp, err := config.ovClient.GetConnectionTemplateByURI(conTempURI)
+			if err != nil {
+				return fmt.Errorf("unable to retrieve connection template: %s", err)
+			}
+			// filter URI
+			id := strings.Split(conTempURI.String(), "/")[3]
```

Comment:
You can use path.Base(subnet.URI.String()) instead of indexing. It fetches the id from the url.

Reply:
done